File: LeetCode/Array/JS/Leetcode1200-minimum-absolute-difference.js
var minimumAbsDifference = function (arr) {
    arr.sort((a, b) => a - b);
    let [res, n, mini] = [[], arr.length, 1 << 30];
    for (let i = 0; i < n - 1; i++) {
        let diff = arr[i + 1] - arr[i];
        if (diff == mini) {
            res.push([arr[i], arr[i + 1]]);
        } else if (diff < mini) {
            mini = diff;
            res = [[arr[i], arr[i + 1]]];
        }
    }
    return res;
};

File: LeetCode/Array/Python/Leetcode1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        n, res, mini = len(arr), [], 1 << 31
        for i in range(n - 1):
            diff = arr[i + 1] - arr[i]
            if diff == mini:
                res.append([arr[i], arr[i + 1]])
            elif diff < mini:
                mini = diff
                res = [[arr[i], arr[i + 1]]]
        return res
